MozillaFoundation/foundation.mozilla.org
# Localize related blog topics

# Description

This PR will ensure that the routes are building from the `parent_or_root` of the active locale

Link to sample test page: https://foundation-s-tp1-770-12-bkm9ye.mofostaging.net/en/blog/dating-app-user-privacy/
Related PRs/issues: TP1-770 #12445

# To Test
1) Review the test page
2) Verify Blog Topic links in the `Related content` section are linking to the `en` localized blog topic pages
3) Switch `/en/` locale to `/fr/`
4) Verify the Blog Topic links in the `Related content` section are linking to the `fr` localized blog topic pages

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-807)
